Apalis imx8 spl conversion

igoropaniuk opened the pull request:
Please do not submit a Pull Request via github.  Our project makes use of
mailing lists for patch submission and review.  For more details please
see https://www.denx.de/wiki/U-Boot/Patches

MrCry0 reviewed (changes requested):
I would split this commit into 3 parts:
- make required nodes available in SPL.
- fix SPL serial output issue.
- fix including the *u-boot.dtsi file to keep the platform dts file closer to the kernel variant.

Tim-Anderson reviewed (approved):
LGTM

ricardosalveti commented:
> I would split this commit into 3 parts:
> 
> * make required nodes available in SPL.
> * fix SPL serial output issue.
> * fix including the *u-boot.dtsi file to keep the platform dts file closer to the kernel variant.

I would just add that we might want to split the SPL enablement patch into 2 as well, one for the ifdef logic and another just for the config changes, as it is then easier to rebase changes in the future.

igoropaniuk committed:
[FIO toup] apalis-imx8: add support of SPL build

Sync with NXP reference board iMX8QM-MEK, where SPL build is supported.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk committed:
[FIO toup] apalis-imx8: enable SPL build

Enable SPL build by default.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk committed:
[FIO toup] ARM: dts: fsl-imx8qm-apalis: don't include u-boot.dtsi

Don't include explicitly fsl-imx8qm-apalis-u-boot.dtsi, as it's done
automatically by dtc, which also permits to add additional properties
to nodes, defined in board dts file after this include.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk committed:
[FIO toup] ARM: dts: fsl-imx8qm-apalis: use dm-spl properties

Use u-boot,dm-spl property for nodes, that should be accessible by
SPL instead of u-boot,dm-pre-proper.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk committed:
[FIO toup] ARM: dts: fsl-imx8qm-apalis: add missing nodes for spl

Add additional nodes to u-boot.dtsi, that fix SPL serial output issues.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

igoropaniuk commented:
all comments addressed

MrCry0 reviewed (commented):
Could you please swap **"enable SPL build"** and **"add support"** commits? It would have seemed more natural.
4aedff9 <-> 5575019

ricardosalveti reviewed (approved):
LGTM, I would just change the FIO tag to be 'toup' instead of 'internal' as you said we plan on upstreaming the changes.

mike-scott commented on configs/apalis-imx8_defconfig:
```diff
@@ -1,16 +1,31 @@
 CONFIG_ENV_SIZE=0x2000
 CONFIG_ENV_OFFSET=0xFFFFDE00
 CONFIG_DM_GPIO=y
+CONFIG_SPL_LOAD_IMX_CONTAINER=y
```
Are we sure we need this?
```
config SPL_LOAD_IMX_CONTAINER
        bool "Enable SPL loading U-Boot as a i.MX Container image"
        depends on SPL
        help
          This is to let SPL could load i.MX8 Container image
```

mike-scott replied:
(Discussion from Slack: this is the "default" defconfig.  We disable this for LmP where we use u-boot.itb)

mike-scott commented on board/toradex/apalis-imx8/apalis-imx8.c:
```diff
@@ -238,7 +239,9 @@ static void select_dt_from_module_version(void)
 }
+#endif
 
+#ifndef CONFIG_SPL_BUILD
```
Seems like we can drop both of these additions, which would let the `#ifndef` block continue?

mike-scott commented on board/toradex/apalis-imx8/apalis-imx8.c:
```diff
@@ -268,7 +272,7 @@ int board_init(void)
 }
 /* todo: With that function in ther is no console output in linux, drop for now */
-#if 0
+#ifdef CONFIG_SPL_BUILD
```
We might want to fix the todo comment above to reflect the new adjustment (as it's not entirely dropped now):
- We release the UART in the SPL hand-off
- But don't release it do to bug in the u-boot proper hand off

mike-scott commented:
Looks good.  I only have 2 minor nits.

mike-scott commented on include/configs/apalis-imx8.h:
```diff
@@ -9,14 +9,44 @@
+
+#define CONFIG_SPL_LDSCRIPT		"arch/arm/cpu/armv8/u-boot-spl.lds"
+#define CONFIG_SPL_STACK		0x013fff0
+#define CONFIG_SPL_BSS_START_ADDR      0x00130000
```
Are these tabs or spaces?   This whole block of #define look a bit messy.

igoropaniuk replied:
fixed

mike-scott reviewed (approved):
LGTM
